DigitalLearningSolutions.Web/Controllers/MyAccountController.cs

```diff
@@ -49,23 +56,87 @@ public IActionResult EditDetails()
         }
 
         [HttpPost]
-        public IActionResult EditDetails(EditDetailsViewModel model)
+        public IActionResult EditDetails(EditDetailsViewModel model, string action)
+        {
+            switch (action)
+            {
+                case "save":
+                    return EditDetailsPostSave(model);
+                case "previewImage":
+                    return EditDetailsPostPreviewImage(model);
+                case "removeImage":
+                    return EditDetailsPostRemoveImage(model);
+                default:
+                    return View(model);
+            }
+        }
+
+        private IActionResult EditDetailsPostSave(EditDetailsViewModel model)
         {
             if (!ModelState.IsValid)
             {
                 return View(model);
             }
 
+            if (model.ProfilePicture != null)
+            {
+                ModelState.AddModelError("ProfilePicture", "Preview your new profile picture before saving");

```

Comment:
Would it be better to do this for them automatically (i.e. return `EditDetailsPostPreviewImage` with this error) so they don't have to click preview and submit the form again?

Reply:
Re UX, I think it makes sense not to auto-preview on submit, because otherwise the user will see the error message but also an image preview, which contradicts the error message. If I get a form error, I'd expect to take an action before submission based on the message. 🙏 

Also, HEE was happy with the behaviour, so I'm happy to leave it.